taxonomy-editor-frontend/src/pages/search/results.jsx

```diff
@@ -0,0 +1,231 @@
+                </Grid>
+                <Typography variant="h6" sx={{mt: 2, mb: 1}}>
+                Number of nodes found: {nodes.length}
+                </Typography>
+                {/* Table for listing all nodes in taxonomy */}
+                <TableContainer sx={{width: 375}} component={Paper}>
+                    <Table>
+                        <TableHead>
+                        <TableRow>
+                            <Stack direction="row" alignItems="center">
+                                <TableCell align="left">
+                                <Typography variant="h6">
+                                    Nodes
+                                </Typography>
+                                </TableCell>
+                                <IconButton sx={{ml: 1, color: "#808080"}} onClick={handleOpenAddDialog}>
+                                    <AddBoxIcon />
+                                </IconButton>
+                            </Stack>
+                            <TableCell align="left">
+                            <Typography variant="h6">
+                                Action
+                            </Typography>
+                            </TableCell>
+                        </TableRow>
+                        </TableHead>
+                        <TableBody>
+                            {nodes.map((nodeID) => (
+                                <TableRow
+                                key={nodeID}
+                                >
+                                    <TableCell align="left" component="td" scope="row">
+                                        <Typography variant="subtitle1">
+                                            {nodeID}
+                                        </Typography>
+                                    </TableCell>
+                                    <TableCell align="left" component="td" scope="row">
+                                        <IconButton onClick={event => handleClick(event, nodeID) } aria-label="edit">
+                                            <EditIcon color="primary"/>
+                                        </IconButton>
+                                    </TableCell>
+                                </TableRow>
+                            ))}
+                        </TableBody>
+                    </Table>
+                </TableContainer>
+                {/* Dialog box for adding nodes */}
+                <Dialog open={openAddDialog} onClose={handleCloseAddDialog}>
+                    <DialogTitle>Add a node</DialogTitle>
+                    <DialogContent>
+                    <Stack direction="row" alignItems="center" sx={{mt: 1, ml: 2, mr: 2, mb: 2}}>
+                        <Typography>Type of node</Typography>
+                        <FormControl sx={{ml: 6.5}}>
+                            <InputLabel>Type</InputLabel>
+                            <Select
```

Comment:
if this is not ready I think we can just not include it in the PR

Reply:
Ah, it technically isn't part of the MVP. So, a comment has been mentioned and an issue will be opened for the same to be implemented in the future.

Reply:
if not part of MVP I think is ok to remove it from the code and add an issue. Better not have things that are to be implemented but are shipped.

Reply:
This code handles the addition of root nodes which is an "entry" in a taxonomy, which is an important feature and is part of the MVP. Hence, I think it shouldn't be removed since this feature is a vital part of the editor.

In the future, this same dialog box must help in adding stopwords and synonyms (another  type of node) in the Taxonomy. So, a comment for that and an issue will be created.